add basic support files for rtl87x2g series soc

YuzhuoLiuRTK opened the pull request:
In this PR, I add code for supporting the realtek rtl87x2g series soc.
With this PR, you can build and run a hello_world app with rtl8762gn_evb.
You can also use drivers such as urat, dma, gpio, adc and counter with rtl87x2g series soc.
Hi @RSGraceGuan @ZhiyuanTang17 , Please help to review this PR.

ZhiyuanTang17 committed:
dts: add dts support for rtl87x2g

ZhiyuanTang17 committed:
boards: rtl8762gn_evb: add rtl8762gn_evb support

ZhiyuanTang17 committed:
soc: rtk_bee: add rtl87x2g series soc support

ZhiyuanTang17 committed:
drivers: support clock-ctrl, pinctrl and uart of rtl87x2g

ZhiyuanTang17 committed:
soc: rtl87x2g: disable interrupt during vector table update

ZhiyuanTang17 committed:
drivers: clock_control: rename APBPeriph_RTZIP_CLOCK for rtl87x2g

YuzhuoLiuRTK committed:
drivers: cctl: add descriptions for rtl87x2g cctl binding file

Signed-off-by: liuyuzhuo <yuzhuo_liu@realsil.com.cn>

YuzhuoLiuRTK committed:
drivers: pinctrl: add descriptions for rlt87x2g pinctrl binding file

Signed-off-by: liuyuzhuo <yuzhuo_liu@realsil.com.cn>

YuzhuoLiuRTK committed:
drivers: uart: modify comments for rtl87x2g uart driver

Signed-off-by: liuyuzhuo <yuzhuo_liu@realsil.com.cn>

YuzhuoLiuRTK committed:
drivers: uart: rename CONFIG_RTK to CONFIG_REALTEK in kconfig file for rtl87x2g

Signed-off-by: liuyuzhuo <yuzhuo_liu@realsil.com.cn>

YuzhuoLiuRTK committed:
drivers: gpio: add gpio driver for rtl87x2g

YuzhuoLiuRTK committed:
tests: drivers: gpio_basic_api: add overlay for rtl8762gn_evb

YuzhuoLiuRTK committed:
drivers: dma: add dma driver for rtl87x2g

YuzhuoLiuRTK committed:
tests: drivers: dma: loop_transfer: add rtl8762gn_evb overlay and conf

YuzhuoLiuRTK committed:
tests: drivers: uart_async_api: add rtl8762gn_evb overlay

YuzhuoLiuRTK committed:
drivers: adc: add adc driver for rtl87x2g

YuzhuoLiuRTK committed:
tests: drivers: adc_api: add rtl8762gn_evb overlay

YuzhuoLiuRTK committed:
drivers: counter: add counter driver for rtl87x2g

YuzhuoLiuRTK committed:
tests: drivers: counter_basic_api: add rtl8762gn_evb overlay

YuzhuoLiuRTK committed:
Kconfig: rename RTK to REALTEK

YuzhuoLiuRTK committed:
drivers: pwm: add pwm driver for rtl87x2g

YuzhuoLiuRTK committed:
tests: drivers: pwm_api: add rtl8762gn_evb overlay

ZhiyuanTang17 commented on boards/arm/rtl8762gn_evb/rtl8762gn_evb.dts:
```diff
@@ -0,0 +1,55 @@
+/dts-v1/;
+#include <rtk/rtl8762gn_evb.dtsi>
```
Please change the dts/arm/rtk/rtl8762gn_evb.dtsi to dts/arm/**realtek**/rtl8762gn_evb.dtsi

YuzhuoLiuRTK replied:
Done

ZhiyuanTang17 commented on boards/arm/rtl8762gn_evb/rtl8762gn_evb.dts:
```diff
@@ -0,0 +1,55 @@
+
+/ {
+	model = "RealTek rtl8762gn evb ";
+	compatible = "rtk,rtl8762gn_evb";
```
Please change the "rtk,rtl8762gn_evb" to"realtek,rtl8762gn_evb"

YuzhuoLiuRTK replied:
Done, please help review

ZhiyuanTang17 reviewed (commented):
I test hello_world app with this PR https://github.com/rtkconnectivity/hal_realtek/pull/4 
And hello_world app works well.
There are some places need to change rtk to realtek，please help to modify.

YuzhuoLiuRTK committed:
drivers: sensors: aon_qdec_rtl87x2g: add qdec driver for rtl87x2g

YuzhuoLiuRTK committed:
sampless: sensor: qdec: add rtl8762gn_evb overlay

YuzhuoLiuRTK committed:
dts: rename rtk to realtek

ZhiyuanTang17 reviewed (approved):
I am ok with this PR merged. Please let @RSGraceGuan help to review.

YuzhuoLiuRTK committed:
drivers: spi: add spi driver for rtl87x2g

YuzhuoLiuRTK committed:
tests: drivers: spi_loopback: add rtl8762gn_evb overlay and conf

YuzhuoLiuRTK committed:
drivers: kscan: add kscan driver for rtl87x2g

YuzhuoLiuRTK committed:
samples: drivers: kscan: add rtl8762gn_evb overlay

YuzhuoLiuRTK committed:
drivers: i2c: add i2c driver for rtl87x2g

YuzhuoLiuRTK committed:
drivers: gpio: add gpio debounce time configure for rtl87x2g

Add vendor flag "RTL87X2G_GPIO_INPUT_DEBOUNCE_MS" for dts and gpio_pin_configure api to config debounce time.

Signed-off-by: liuyuzhuo <yuzhuo_liu@realsil.com.cn>

YuzhuoLiuRTK committed:
drivers: uart: poll tx_fifo_len in poll_out api

Use tx_fifo_len < tx_fifo_size condition instead of tx_fifo_empty
condition to increase polling efficiency.

Signed-off-by: liuyuzhuo <yuzhuo_liu@realsil.com.cn>

YuzhuoLiuRTK committed:
drivers: kscan: add software debounce for rtl87x2g

Check scan result for specificed times to filter button press jitter.
Add all release interrupt to filter button release jitter.

Signed-off-by: liuyuzhuo <yuzhuo_liu@realsil.com.cn>

YuzhuoLiuRTK committed:
drivers: adc: add bypass mode and get voltage for rtl87x2g
Add properties "is-bypass-mode" in binding file and add releated code in
c file for users to configure bypass mode in dts. Call ADC_GetVoltage
api in adc_read api to convert raw sample data to actual voltage for
upper use.

Signed-off-by: liuyuzhuo <yuzhuo_liu@realsil.com.cn>

YuzhuoLiuRTK committed:
drivers: counter: modify prescaler value for rtl87x2g

Signed-off-by: liuyuzhuo <yuzhuo_liu@realsil.com.cn>

YuzhuoLiuRTK committed:
drivers: clock_control: rename 24badc for rtl87x2g

Signed-off-by: liuyuzhuo <yuzhuo_liu@realsil.com.cn>

YuzhuoLiuRTK committed:
drivers: gpio: fix flase trigger for rtl87x2g

add delay flow after configure gpio trigger mode to avoid false trigger

Signed-off-by: liuyuzhuo <yuzhuo_liu@realsil.com.cn>
